cms_helpers/cms_toolbars.py

```diff
@@ -24,18 +26,32 @@ def populate(self):
         if not current_page_menu or not self.page:
             return
 
+        languages = get_language_list(self.current_site.id)
+        is_single_lang = len(languages) < 2
         position = self.get_item_position(current_page_menu)
-
         urls = self.get_title_extension_admin()
+        page = self._get_page()
+        titleset = page.title_set.filter(language__in=languages)
 
         if hasattr(self.toolbar, 'edit_mode_active'):
             not_edit_mode = not self.toolbar.edit_mode_active
         else:
             not_edit_mode = not self.toolbar.edit_mode
 
-        for title_extension, url in urls:
-            current_page_menu.add_modal_item(
-                self.model._meta.verbose_name,
-                url=url, position=position,
-                disabled=not_edit_mode,
-            )
+        extended_menu = current_page_menu if is_single_lang else (
+            current_page_menu.get_or_create_menu(
+                key='{0}_menu'.format(self.model._meta.db_table),
+                verbose_name=self.model._meta.verbose_name,
+                position=position, disabled=not_edit_mode))
+
+        nodes = [(title_extension, url, title) for (
+            (title_extension, url), title) in zip(urls, titleset)]
```

Comment:
Is this merge save as in the ordering will always be correct on both sides?

Reply:
hmm ... this is what they advise in django-cms docu. I added tests and it works, but I see the problem. Unfortunately we don't have language in url. Do you think we should take obj id from url and hit the db instead of this merge?

Reply:
If this is the "blessed way", I'm fine with it.